webapp/src/components/Bid/Bid.js

```diff
@@ -134,72 +133,91 @@ export default class Bid extends React.PureComponent {
-              scale={1}
-              className="mortgage-amount-icon"
-            />
-          </Grid.Column>
-          <Grid.Column width={showAssetDetail ? 3 : 3}>
-            <h3>{t('global.time_left')}</h3>
-            <p>{distanceInWordStrict(parseInt(bid.expires_at, 10))}</p>
-          </Grid.Column>
-          <Grid.Column
-            width={isOwner && showAssetDetail ? 3 : 6}
-            className={'actions'}
-          >
-            {!isOwner &&
-              !hasSameSellerAndBidder && (
-                <Button onClick={preventDefault(onUpdate)}>
-                  {t('global.update')}
-                </Button>
+          <Grid.Column width={showAssetDetail ? 13 : 16}>
+            <Grid className="inner-grid">
+              {showAssetDetail && (
+                <Grid.Column computer={4} tablet={4} mobile={8}>
+                  <h3>{t('bid.bid_asset')}</h3>
+                  <Link className="asset-name" to={assetDetailLink}>
+                    {this.renderAssetData()}
+                  </Link>
+                </Grid.Column>
               )}
-            <Button onClick={preventDefault(onConfirm)}>
-              {!isOwner || hasSameSellerAndBidder
-                ? t('global.cancel')
-                : t('global.accept')}
-            </Button>
+              {isOwner || !showAssetDetail ? (
+                <Grid.Column
+                  computer={showAssetDetail ? 4 : 3}
+                  tablet={showAssetDetail ? 4 : 3}
+                  mobile={showAssetDetail ? 8 : 16}
+                >
+                  <h3>{t('global.from')}</h3>
+                  <div className="address-wrapper" title={bid.bidder}>
+                    <Link to={locations.profilePageDefault(bid.bidder)}>
+                      <AddressBlock
+                        address={bid.bidder}
+                        scale={3.3}
+                        hasTooltip={false}
+                        hasLink={false}
+                      />&nbsp;
+                      <span className="short-address">
+                        {hasSameSellerAndBidder
+                          ? t('global.you')
+                          : shortenOwner(bid.bidder)}
+                      </span>
+                    </Link>
+                  </div>
+                </Grid.Column>
+              ) : null}
+              <Grid.Column
+                computer={showAssetDetail ? 4 : 3}
+                tablet={showAssetDetail ? 4 : 3}
+                mobile={showAssetDetail ? 8 : 16}
+              >
+                <h3>{t('global.price')}</h3>
+                <Mana
+                  amount={bid.price}
+                  size={15}
+                  scale={1}
+                  className="mortgage-amount-icon"
+                />
+              </Grid.Column>
+              <Grid.Column
+                computer={showAssetDetail ? 4 : 3}
+                tablet={showAssetDetail ? 4 : 3}
+                mobile={showAssetDetail ? 8 : 16}
```

Comment:
This seems to be repeated thrice, maybe we could do something like:

```js
const sizeByResolution = {
  computer: showAssetDetail ? 4 : 3
  tablet: showAssetDetail ? 4 : 3
  mobile: showAssetDetail ? 8 : 16
}
```

and then

```jsx
<Component
  {...sizeByResolution}
 />
```

Reply:
You are right! that is better